modal/_utils/async_utils.py

```diff
@@ -544,3 +553,117 @@ async def producer(producer_id: int, iterable: Union[AsyncIterable[T], Iterable[
 
 async def callable_to_agen(awaitable: Callable[[], Awaitable[T]]) -> AsyncGenerator[T, None]:
     yield await awaitable()
+
+
+async def async_map(
+    input_iterable: Union[AsyncIterable[T], Iterable[T]],
+    async_mapper_func: Callable[[T], Awaitable[V]],
+    concurrency: int,
+) -> AsyncGenerator[V, None]:
+    input_queue: asyncio.Queue[Tuple[str, Union[T, None]]] = asyncio.Queue()
+    output_queue: asyncio.Queue[Tuple[str, Union[V, Exception]]] = asyncio.Queue()
+    output_event = asyncio.Event()
```

Comment:
Just realized we never do `output_event.set()`! 💥 
Pretty bad story for the client test coverage that pretty much all tests pass anyways 😬 

Btw, I just realized that since we have to reset the `output_event.wait()` task in each iteration of the loop anyways, we might as well just listen for `output_queue.get()` instead there and skip this event completely